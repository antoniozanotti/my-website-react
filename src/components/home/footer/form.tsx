"use client";
import React, { useRef, useState } from "react";
import { TzButton, TzInput, TzTextarea } from "topaz-react";
import emailjs from "@emailjs/browser";
import ReCAPTCHA from "react-google-recaptcha";

export default function Form() {
  const form = useRef<HTMLFormElement>(null);
  const fromName = useRef<HTMLInputElement>(null);
  const [fromNameValue, setFromNameValue] = useState("");
  const [status, setStatus] = useState("typing");
  const recaptchaRef = useRef<ReCAPTCHA>(null);
  const [recaptchaValue, setRecaptchaValue] = useState("");

  const onChangeRecaptcha = (token: string | null) => {
    //const token = recaptchaRef.current?.getValue() ?? "";
    setRecaptchaValue(token ?? "");
  };

  const handleSubmit = (e: React.SyntheticEvent) => {
    e.preventDefault();
    recaptchaRef.current?.execute();
    recaptchaRef.current?.reset();
    if (recaptchaValue == "") {
      return false;
    }
    setStatus("submitting");
    setFromNameValue(fromName.current?.value ?? "");
    emailjs
      .sendForm("service_7vfby3r", "template_a9dy2ok", form.current ?? "", {
        publicKey: process.env.NEXT_PUBLIC_EMAILJS_PUBLIC_KEY,
      })
      .then(
        () => {
          setStatus("success");
          form.current?.reset();
        },
        (error) => {
          setStatus("error");
        }
      );
  };

  const recaptchaKey = process.env.NEXT_PUBLIC_GOOGLE_RECAPTCHA ?? "";

  return (
    <form
      id="form_lets_work_together"
      className="flex flex-col space-y-4"
      ref={form}
      onSubmit={handleSubmit}
    >
      <label className="text-primary-2 dark:text-dark-primary-2 flex flex-col space-y-1">
        <span className="px-[10px] sm:px-[14px] lg:px-[20px]">Name</span>
        <TzInput
          type="text"
          variant="secondary"
          name="fromName"
          required
          ref={fromName}
        />
      </label>
      <label className="text-primary-2 dark:text-dark-primary-2 flex flex-col space-y-1">
        <span className="px-[10px] sm:px-[14px] lg:px-[20px]">Email</span>
        <TzInput type="email" variant="secondary" name="fromEmail" required />
      </label>
      <label className="text-primary-2 dark:text-dark-primary-2 flex flex-col space-y-1">
        <span className="px-[10px] sm:px-[14px] lg:px-[20px]">Message</span>
        <TzTextarea rows={7} variant="secondary" name="message" required />
      </label>
      <ReCAPTCHA
        ref={recaptchaRef}
        sitekey={recaptchaKey}
        size="invisible"
        onChange={onChangeRecaptcha}
      />
      {status === "success" && (
        <p>
          Hi {fromNameValue}, I will respond within 24 hours.
          <br />
          Thank you for the message sent.
        </p>
      )}
      {status === "error" && (
        <p>Hi {fromNameValue}, something went wrong. Please try later..</p>
      )}
      <div className="flex place-content-end">
        <TzButton
          iconName="ArrowRightIcon"
          isIconAfterLabel
          variant="accent"
          label="Submit"
        />
      </div>
      <input type="hidden" name="g-recaptcha-response" value={recaptchaValue} />
    </form>
  );
}
