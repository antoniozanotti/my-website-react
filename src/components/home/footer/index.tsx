"use client";
import { TzButton, TzIcon } from "topaz-react";
import Section from "@/components/ui/section";
import Form from "./form";
import SocialMedia from "@/components/common/social-media";
import ImgLookingUpXl from "@images/antonio-zanotti-looking-up-xl.jpg"
import ImgLookingUpLg from "@images/antonio-zanotti-looking-up-lg.jpg"
import ImgLookingUpMd from "@images/antonio-zanotti-looking-up-md.jpg"
import ImgLookingUpSm from "@images/antonio-zanotti-looking-up-sm.jpg"
import ImgLookingUpDefault from "@images/antonio-zanotti-looking-up-default.jpg"

export default function Index() {
  return (
    <footer>
      <Section title="Let's work together">
        <div className="lg:grid lg:grid-cols-2 sm:gap-x-8 lg:gap-x-16 mt-4">
          <div className="pb-12">
            <p>
              I would like to be part of a team whose people respect each other,
              listen to each other and who work together to achieve clear goals. Leave a message, thank you in advance for your
              attention.
            </p>
            <picture className="rounded block overflow-hidden mt-4">
              <source media="(min-width: 1280px)" srcSet={ImgLookingUpXl.src} />
              <source media="(min-width: 1024px)" srcSet={ImgLookingUpLg.src} />
              <source media="(min-width: 768px)" srcSet={ImgLookingUpMd.src} />
              <source media="(min-width: 640px)" srcSet={ImgLookingUpSm.src} />
              <img src={ImgLookingUpDefault.src} alt="Antônio Zanotti looking up" />
            </picture>
            <SocialMedia className="pt-4" />
          </div>
          <div>
            <Form />
          </div>
        </div>
        <div className="pt-12 pb-4 flex place-content-center lg:place-content-end">
          <TzButton
            iconName="ArrowUpIcon"
            variant="secondary"
            onClick={() => {
              window.scroll({
                top: 0,
                left: 0,
                behavior: "smooth",
              });
            }}
          />
        </div>
        <p className="text-center lg:text-left lg:mt-[-50px]">
          Made with{" "}
          <TzIcon iconName="HeartIcon" variant="negative" className="inline" />{" "}
          by Antônio Zanotti
        </p>
      </Section>
    </footer>
  );
}
