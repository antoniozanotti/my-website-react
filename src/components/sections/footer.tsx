import { TzButton, TzIcon } from "topaz-react";
import Section from "@/components/ui/section";
import Form from "@/components/blocks/form";
import SocialMedia from "@/components/blocks/social-media";
import AntonioZanotti2Img from "@images/antonio_zanotti_2.jpg"

export default function Footer() {
  return (
    <footer>
      <Section title="# Let's work together">
        <div className="lg:grid lg:grid-cols-2 sm:gap-x-8 lg:gap-x-16 mt-4">
          <div className="pb-12">
            <Form />
          </div>
          <div className="lg:mt-[27px]">
            <SocialMedia className="pb-4" />
            <img
              src={AntonioZanotti2Img.src}
              alt="Antônio Zanotti"
              className="rounded"
            />
          </div>
        </div>
        <div className="pt-12 pb-4 flex place-content-center lg:place-content-end">
          <TzButton
            iconName="ArrowUpIcon"
            variant="secondary"
            onClick={() => {
              window.scroll({
                top: 0,
                left: 0,
                behavior: "smooth",
              });
            }}
          />
        </div>
        <p className="text-center lg:mt-[-50px]">
          Made with{" "}
          <TzIcon iconName="HeartIcon" variant="negative" className="inline" />{" "}
          by Antônio Zanotti
        </p>
      </Section>
    </footer>
  );
}
