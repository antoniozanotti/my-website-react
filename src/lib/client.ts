import { ApolloClient, HttpLink, InMemoryCache } from "@apollo/client";
import { registerApolloClient } from "@apollo/experimental-nextjs-app-support/rsc";

const TOKEN = process.env.CONTENTFUL_ACCESS_TOKEN;
const SPACE = process.env.CONTENTFUL_SPACE_ID;
const ENVIRONMENT = process.env.CONTENTFUL_ENVIRONMENT_ID;
const URL = `https://graphql.contentful.com/content/v1/spaces/${SPACE}/environments/${ENVIRONMENT}`;

export const { getClient } = registerApolloClient(() => {
  return new ApolloClient({
    cache: new InMemoryCache(),
    link: new HttpLink({
      uri: URL,
      headers: {
        Authorization: `Bearer ${TOKEN}`,
      },
    }),
  });
});
