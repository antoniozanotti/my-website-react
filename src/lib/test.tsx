import React, { ReactElement } from "react";
import {
  render,
  RenderOptions,
  RenderResult,
  Screen,
} from "@testing-library/react";
import { Theme } from "@/domain/theme/theme/theme";
import { Providers } from "../app/providers";
import messages from "@/messages/en.json";

function AllTheProviders({ children }: { children: React.ReactNode }) {
  return (
    <Providers locale="en" messages={messages}>
      <Theme>
        <div data-testid="container">{children}</div>
      </Theme>
    </Providers>
  );
}

function customRender(
  ui: ReactElement,
  options?: Omit<RenderOptions, "wrapper">
): RenderResult {
  return render(ui, { wrapper: AllTheProviders, ...options });
}

function firstComponent(screen: Screen) {
  return screen.getByTestId("container").firstChild;
}

export * from "@testing-library/react";
export * from "@testing-library/user-event";
export { customRender as render, firstComponent };
import "@testing-library/jest-dom";

jest.mock("next/navigation", () => {
  return {
    useRouter() {
      return { push: () => {} };
    },
    useParams: () => ({}),
  };
});
