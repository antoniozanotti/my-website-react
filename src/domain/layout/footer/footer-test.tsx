import { test, expect } from '@playwright/experimental-ct-react';
import { Footer } from "./footer";

test.describe("Footer", () => {
  test("should render footer", async ({ mount }) => {
    const component = await mount(<Footer />);
    await expect(component.locator(":scope:is(footer)")).toBeVisible();
  });
});
