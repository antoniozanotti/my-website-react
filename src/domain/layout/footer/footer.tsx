"use client";

import { useTranslations } from "next-intl";
import { Section } from "@/domain/ui/section/section";
import { SocialMedia } from "@/domain/home/social-media/social-media";
import { Form } from "./form/form";
import { TzButton, TzIcon } from "topaz-react";
import { Picture } from "./picture/picture";

export function Footer() {
  const t = useTranslations("Footer");

  function useGoToTop() {
    window.scroll({
      top: 0,
      left: 0,
      behavior: "smooth",
    });
  }

  return (
    <footer>
      <Section title={t("title")}>
        <div className="lg:grid lg:grid-cols-2 sm:gap-x-8 lg:gap-x-16 mt-4">
          <div className="pb-12">
            <p>{t("intro")}</p>
            <Picture />
            <SocialMedia className="pt-4" />
          </div>
          <div>
            <Form />
          </div>
        </div>
        <div className="pt-12 pb-4 flex place-content-center lg:place-content-end">
          <TzButton
            aria-label="Go to Top Page"
            iconName="ArrowUpIcon"
            variant="secondary"
            onClick={useGoToTop}
          />
        </div>
        <p className="text-center lg:text-left lg:mt-[-50px]">
          {t("made")}{" "}
          <TzIcon iconName="HeartIcon" variant="negative" className="inline" />{" "}
          {t("by")} Ant√¥nio Zanotti
        </p>
      </Section>
    </footer>
  );
}
