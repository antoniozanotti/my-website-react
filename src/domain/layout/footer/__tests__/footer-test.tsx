import { render } from "@/lib/test";
import { Footer } from "../footer";

describe("Footer", () => {
  it("should render footer", async () => {
    const {container} = render(<Footer />);
    const footer = container.querySelector("footer");

    expect(footer).toBeVisible();
  });
});
