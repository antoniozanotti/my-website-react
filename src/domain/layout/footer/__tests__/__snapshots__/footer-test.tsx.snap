// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer should render footer 1`] = `
<div>
  <div
    class="dark"
  >
    <div
      class="className bg-dark-2 dark:bg-dark-dark-2 text-primary dark:text-dark-primary text-c3 lg:text-c4"
    >
      <div
        data-testid="container"
      >
        <footer>
          <section>
            <h2
              class="leading-none text-c9 sm:text-c10 lg:text-c11 text-accent-1 dark:text-dark-accent-1 drop-shadow mt-[164px] "
              id=""
            >
              Let's work together
            </h2>
            <div
              class="lg:grid lg:grid-cols-2 sm:gap-x-8 lg:gap-x-16 mt-4"
            >
              <div
                class="pb-12"
              >
                <p>
                  I would like to be part of a team whose people respect each other, listen to each other and who work together to achieve clear goals. Leave a message, thank you in advance or your attention.
                </p>
                <picture
                  class="rounded block overflow-hidden mt-4"
                >
                  <source
                    media="(min-width: 1280px)"
                    srcset="/img.jpg"
                  />
                  <source
                    media="(min-width: 1024px)"
                    srcset="/img.jpg"
                  />
                  <source
                    media="(min-width: 768px)"
                    srcset="/img.jpg"
                  />
                  <source
                    media="(min-width: 640px)"
                    srcset="/img.jpg"
                  />
                  <img
                    alt="AntÃ´nio Zanotti looking up"
                    src="/img.jpg"
                  />
                </picture>
                <div
                  class="flex place-items-center gap-[30px] pt-4"
                >
                  <a
                    class="hover:underline flex place-items-center gap-[15px]"
                    href="https://www.linkedin.com/in/antoniozanotti/"
                    target="_blank"
                    title="Linked In"
                  >
                    <svg
                      class="w-[38px] h-[38px] lg:w-[48px] lg:h-[48px]"
                      fill="none"
                      role="img"
                      viewBox="0 0 48 48"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <title>
                        Linked In Brand
                      </title>
                      <g
                        clip-path="url(#clip0_44_42)"
                        filter="url(#filter0_i_44_42)"
                      >
                        <path
                          d="M0 3.438C0 1.539 1.578 0 3.525 0H44.475C46.422 0 48 1.539 48 3.438V44.562C48 46.461 46.422 48 44.475 48H3.525C1.578 48 0 46.461 0 44.562V3.438ZM14.829 40.182V18.507H7.626V40.182H14.829ZM11.229 15.546C13.74 15.546 15.303 13.884 15.303 11.802C15.258 9.675 13.743 8.058 11.277 8.058C8.811 8.058 7.2 9.678 7.2 11.802C7.2 13.884 8.763 15.546 11.181 15.546H11.229ZM25.953 40.182V28.077C25.953 27.429 26.001 26.781 26.193 26.319C26.712 25.026 27.897 23.685 29.889 23.685C32.496 23.685 33.537 25.671 33.537 28.587V40.182H40.74V27.75C40.74 21.09 37.188 17.994 32.448 17.994C28.626 17.994 26.913 20.094 25.953 21.573V21.648H25.905C25.9209 21.623 25.9369 21.598 25.953 21.573V18.507H18.753C18.843 20.541 18.753 40.182 18.753 40.182H25.953Z"
                          fill="currentColor"
                        />
                      </g>
                      <defs>
                        <filter
                          color-interpolation-filters="sRGB"
                          filterUnits="userSpaceOnUse"
                          height="52"
                          id="filter0_i_44_42"
                          width="48"
                          x="0"
                          y="0"
                        >
                          <feflood
                            flood-opacity="0"
                            result="BackgroundImageFix"
                          />
                          <feblend
                            in="SourceGraphic"
                            in2="BackgroundImageFix"
                            mode="normal"
                            result="shape"
                          />
                          <fecolormatrix
                            in="SourceAlpha"
                            result="hardAlpha"
                            type="matrix"
                            values="0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 127 0"
                          />
                          <feoffset
                            dy="4"
                          />
                          <fegaussianblur
                            stdDeviation="2"
                          />
                          <fecomposite
                            in2="hardAlpha"
                            k2="-1"
                            k3="1"
                            operator="arithmetic"
                          />
                          <fecolormatrix
                            type="matrix"
                            values="0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0.25 0"
                          />
                          <feblend
                            in2="shape"
                            mode="normal"
                            result="effect1_innerShadow_44_42"
                          />
                        </filter>
                        <clippath
                          id="clip0_44_42"
                        >
                          <rect
                            fill="white"
                            height="48"
                            width="48"
                          />
                        </clippath>
                      </defs>
                    </svg>
                    Linked In
                  </a>
                  <a
                    class="hover:underline flex place-items-center gap-[15px]"
                    href="https://github.com/antoniozanotti"
                    target="_blank"
                    title="GitHub"
                  >
                    <svg
                      class="w-[38px] h-[38px] lg:w-[48px] lg:h-[48px]"
                      fill="none"
                      role="img"
                      viewBox="0 0 48 48"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <title>
                        GitHub Brand
                      </title>
                      <g
                        filter="url(#filter0_i_44_44)"
                      >
                        <path
                          d="M24 0C10.74 0 0 10.74 0 24C0 34.62 6.87 43.59 16.41 46.77C17.61 46.98 18.06 46.26 18.06 45.63C18.06 45.06 18.03 43.17 18.03 41.16C12 42.27 10.44 39.69 9.96 38.34C9.69 37.65 8.52 35.52 7.5 34.95C6.66 34.5 5.46 33.39 7.47 33.36C9.36 33.33 10.71 35.1 11.16 35.82C13.32 39.45 16.77 38.43 18.15 37.8C18.36 36.24 18.99 35.19 19.68 34.59C14.34 33.99 8.76 31.92 8.76 22.74C8.76 20.13 9.69 17.97 11.22 16.29C10.98 15.69 10.14 13.23 11.46 9.93C11.46 9.93 13.47 9.3 18.06 12.39C19.98 11.85 22.02 11.58 24.06 11.58C26.1 11.58 28.14 11.85 30.06 12.39C34.65 9.27 36.66 9.93 36.66 9.93C37.98 13.23 37.14 15.69 36.9 16.29C38.43 17.97 39.36 20.1 39.36 22.74C39.36 31.95 33.75 33.99 28.41 34.59C29.28 35.34 30.03 36.78 30.03 39.03C30.03 42.24 30 44.82 30 45.63C30 46.26 30.45 47.01 31.65 46.77C36.4147 45.162 40.555 42.1 43.4881 38.0153C46.4211 33.9305 47.9992 29.0287 48 24C48 10.74 37.26 0 24 0Z"
                          fill="currentColor"
                        />
                      </g>
                      <defs>
                        <filter
                          color-interpolation-filters="sRGB"
                          filterUnits="userSpaceOnUse"
                          height="52"
                          id="filter0_i_44_44"
                          width="48"
                          x="0"
                          y="0"
                        >
                          <feflood
                            flood-opacity="0"
                            result="BackgroundImageFix"
                          />
                          <feblend
                            in="SourceGraphic"
                            in2="BackgroundImageFix"
                            mode="normal"
                            result="shape"
                          />
                          <fecolormatrix
                            in="SourceAlpha"
                            result="hardAlpha"
                            type="matrix"
                            values="0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 127 0"
                          />
                          <feoffset
                            dy="4"
                          />
                          <fegaussianblur
                            stdDeviation="2"
                          />
                          <fecomposite
                            in2="hardAlpha"
                            k2="-1"
                            k3="1"
                            operator="arithmetic"
                          />
                          <fecolormatrix
                            type="matrix"
                            values="0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0.25 0"
                          />
                          <feblend
                            in2="shape"
                            mode="normal"
                            result="effect1_innerShadow_44_44"
                          />
                        </filter>
                      </defs>
                    </svg>
                    GitHub
                  </a>
                </div>
              </div>
              <div>
                <form
                  class="flex flex-col space-y-4"
                  id="form_lets_work_together"
                >
                  <label
                    class="text-primary-2 dark:text-dark-primary-2 flex flex-col space-y-1"
                  >
                    <span
                      class="px-[10px] sm:px-[14px] lg:px-[20px]"
                    >
                      Name
                    </span>
                    <input
                      class="rounded focus:outline focus:outline-offset-2 focus:outline-2 h-[28px] sm:h-[38px] lg:h-[48px] px-[10px] sm:px-[14px] lg:px-[20px]  bg-secondary dark:bg-dark-secondary hover:bg-secondary-1 active:bg-secondary-2 focus:outline-secondary dark:hover:bg-dark-secondary-1 dark:active:bg-dark-secondary-2 dark:focus:outline-dark-secondary text-inverse-secondary dark:text-inverse-dark-secondary"
                      name="fromName"
                      required=""
                      type="text"
                      variant="secondary"
                    />
                  </label>
                  <label
                    class="text-primary-2 dark:text-dark-primary-2 flex flex-col space-y-1"
                  >
                    <span
                      class="px-[10px] sm:px-[14px] lg:px-[20px]"
                    >
                      Email
                    </span>
                    <input
                      class="rounded focus:outline focus:outline-offset-2 focus:outline-2 h-[28px] sm:h-[38px] lg:h-[48px] px-[10px] sm:px-[14px] lg:px-[20px]  bg-secondary dark:bg-dark-secondary hover:bg-secondary-1 active:bg-secondary-2 focus:outline-secondary dark:hover:bg-dark-secondary-1 dark:active:bg-dark-secondary-2 dark:focus:outline-dark-secondary text-inverse-secondary dark:text-inverse-dark-secondary"
                      name="fromEmail"
                      required=""
                      type="email"
                      variant="secondary"
                    />
                  </label>
                  <label
                    class="text-primary-2 dark:text-dark-primary-2 flex flex-col space-y-1"
                  >
                    <span
                      class="px-[10px] sm:px-[14px] lg:px-[20px]"
                    >
                      Message
                    </span>
                    <textarea
                      class="rounded focus:outline focus:outline-offset-2 focus:outline-2 h-[177px] px-[10px] sm:px-[14px] lg:px-[20px] py-[7px] sm:py-[10px] lg:py-[12px]  bg-secondary dark:bg-dark-secondary hover:bg-secondary-1 active:bg-secondary-2 focus:outline-secondary dark:hover:bg-dark-secondary-1 dark:active:bg-dark-secondary-2 dark:focus:outline-dark-secondary text-inverse-secondary dark:text-inverse-dark-secondary"
                      name="message"
                      required=""
                      rows="7"
                      variant="secondary"
                    />
                  </label>
                  <div />
                  <div
                    class="flex place-content-end"
                  >
                    <button
                      class="rounded-full font-medium flex gap-x-2 justify-center place-items-center focus:outline focus:outline-offset-2 focus:outline-2  bg-accent dark:bg-dark-accent hover:bg-accent-1 active:bg-accent-2 focus:outline-accent dark:hover:bg-dark-accent-1 dark:active:bg-dark-accent-2 dark:focus:outline-dark-accent text-inverse-accent dark:text-inverse-dark-accent h-[38px] lg:h-[48px] min-w-[38px] lg:min-w-[48px] px-[27px] lg:px-[36px] "
                    >
                      <svg
                        aria-hidden="true"
                        class="w-[16px] h-[16px] lg:w-[20px] lg:h-[20px] text-inverse-accent dark:text-inverse-dark-accent order-last"
                        data-slot="icon"
                        fill="none"
                        stroke="currentColor"
                        stroke-width="1.5"
                        viewBox="0 0 24 24"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <title>
                          ArrowRightIcon
                        </title>
                        <path
                          d="M13.5 4.5 21 12m0 0-7.5 7.5M21 12H3"
                          stroke-linecap="round"
                          stroke-linejoin="round"
                        />
                      </svg>
                      <span>
                        Submit
                      </span>
                    </button>
                  </div>
                </form>
              </div>
            </div>
            <div
              class="pt-12 pb-4 flex place-content-center lg:place-content-end"
            >
              <button
                aria-label="Go to Top Page"
                class="rounded-full font-medium flex gap-x-2 justify-center place-items-center focus:outline focus:outline-offset-2 focus:outline-2  bg-secondary dark:bg-dark-secondary hover:bg-secondary-1 active:bg-secondary-2 focus:outline-secondary dark:hover:bg-dark-secondary-1 dark:active:bg-dark-secondary-2 dark:focus:outline-dark-secondary text-inverse-secondary dark:text-inverse-dark-secondary h-[38px] lg:h-[48px] min-w-[38px] lg:min-w-[48px] "
              >
                <svg
                  aria-hidden="true"
                  class="w-[16px] h-[16px] lg:w-[20px] lg:h-[20px] text-inverse-secondary dark:text-inverse-dark-secondary "
                  data-slot="icon"
                  fill="none"
                  stroke="currentColor"
                  stroke-width="1.5"
                  viewBox="0 0 24 24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <title>
                    ArrowUpIcon
                  </title>
                  <path
                    d="M4.5 10.5 12 3m0 0 7.5 7.5M12 3v18"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                  />
                </svg>
              </button>
            </div>
            <p
              class="text-center lg:text-left lg:mt-[-50px]"
            >
              Made with
               
              <svg
                aria-hidden="true"
                class="w-[16px] h-[16px] lg:w-[20px] lg:h-[20px] text-negative dark:text-dark-negative inline"
                data-slot="icon"
                fill="none"
                stroke="currentColor"
                stroke-width="1.5"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <title>
                  HeartIcon
                </title>
                <path
                  d="M21 8.25c0-2.485-2.099-4.5-4.688-4.5-1.935 0-3.597 1.126-4.312 2.733-.715-1.607-2.377-2.733-4.313-2.733C5.1 3.75 3 5.765 3 8.25c0 7.22 9 12 9 12s9-4.78 9-12Z"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                />
              </svg>
               
              by
               AntÃ´nio Zanotti
            </p>
          </section>
        </footer>
      </div>
    </div>
  </div>
</div>
`;
